enhancements/cert-manager-proxy/http01_challenge_proxy.md

```diff
@@ -0,0 +1,364 @@
+- Redirect filtered HTTP traffic from the API endpoint (`api.cluster.example.com`) on port 80 to the OpenShift Ingress Routers  
+- Use `nftables` for traffic redirection from `API:80` to `PROXY:8888`
+- Handle lifecycle management, upgrades, and configuration of the proxy
+
+The proxy will ensure compatibility with various OCP topologies, including SNO, MNO, and Compact clusters, addressing the challenges of HTTP01 validation for the API endpoint.
+
+### API Extensions
+
+Rather than creating a new CRD, the HTTP01 Challenge Proxy configuration will be added as a new field to the existing **APIServer** CRD in the [openshift/api](https://github.com/openshift/api) repo. This approach aligns with the pattern used by other API server features like audit, encryption, and TLS configuration.
+
+The configuration will be added to the `APIServerSpec` struct and managed by the cluster-kube-apiserver-operator as part of its normal reconciliation loop. Since the APIServer CRD is already a singleton (named "cluster"), no additional enforcement mechanisms are needed.
+
+**API Changes to config/v1/types_apiserver.go:**
+
+```go
+type APIServerSpec struct {
+    // ... existing fields ...
+    
+    // http01ChallengeProxy contains configuration for the HTTP01 challenge proxy
+    // that redirects traffic from the API endpoint on port 80 to ingress routers.
+    // This enables cert-manager to perform HTTP01 ACME challenges for API endpoint certificates.
+    // +optional
+    HTTP01ChallengeProxy *HTTP01ChallengeProxySpec `json:"http01ChallengeProxy,omitempty"`
+}
+
+type HTTP01ChallengeProxySpec struct {
+    // enabled controls whether the HTTP01 challenge proxy is active.
+    // When enabled, the proxy redirects HTTP traffic from API:80 to ingress routers.
+    // +optional
+    Enabled bool `json:"enabled"`
+    
+    // internalPort specifies the internal port used by the proxy service.
+    // +kubebuilder:validation:Minimum=1024
+    // +kubebuilder:validation:Maximum=65535
+    // +optional
+    InternalPort int32 `json:"internalPort,omitempty"`  // default: 8888
+}
+```
+
+**Example Configuration:**
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: APIServer
+metadata:
+  name: cluster
+spec:
+  # existing fields...
+  http01ChallengeProxy:
+    enabled: true
+    internalPort: 8888
+status:
+  # existing status fields...
+  conditions:
+    - type: HTTP01ChallengeProxyReady
+      status: "True"
+      lastTransitionTime: "2025-05-12T00:00:00Z"
+      reason: "Initialized"
+      message: "HTTP01ChallengeProxy is ready"
+```
+    message: "Only one HTTP01ChallengeProxy instance named 'cluster' is allowed per cluster"
+```
+
+This design ensures that:
+- Only one `HTTP01ChallengeProxy` instance can exist cluster-wide
+- The instance must be named `cluster` to enforce the singleton pattern
+- The ValidatingAdmissionPolicy prevents creation of additional instances
```

Comment:
Where did we specify in the document that a VAP would be included?

Could we get an outline of the VAP we expect to be created?